src/annotator/config/test/index-test.js

```diff
@@ -51,24 +67,117 @@ describe('annotator/config/index', function () {
-        sinon.match({ allowInBrowserExt: true })
-      );
+  describe('browser extension', () => {
+    context('browser extension set to true', function () {
+      it('returns only config values where `allowInBrowserExt` is true or the defaultValue if provided', () => {
+        fakeIsBrowserExtension.returns(true);
+        const config = getConfig('all', 'WINDOW');
+        assert.deepEqual(
+          {
+            appType: 'fakeValue',
+            annotations: 'fakeValue',
+            branding: null,
+            clientUrl: 'fakeValue',
+            enableExperimentalNewNoteButton: null,
+            externalContainerSelector: null,
+            focus: null,
+            group: null,
+            notebookAppUrl: 'fakeValue',
+            onLayoutChange: null,
+            openSidebar: true, // coerced
+            query: 'fakeValue',
+            requestConfigFromFrame: null,
+            services: null,
+            showHighlights: null,
+            sidebarAppUrl: 'fakeValue',
+            subFrameIdentifier: 'fakeValue',
+            theme: null,
+            usernameUrl: null,
+          },
+          config
+        );
+      });
+    });
+
+    context('browser extension set to false', function () {
+      it('returns all config values', () => {
+        fakeIsBrowserExtension.returns(false);
+        const config = getConfig('all', 'WINDOW');
+        assert.deepEqual(
+          {
+            appType: 'fakeValue',
+            annotations: 'fakeValue',
+            branding: 'fakeValue',
+            clientUrl: 'fakeValue',
+            enableExperimentalNewNoteButton: 'fakeValue',
+            externalContainerSelector: 'fakeValue',
+            focus: 'fakeValue',
+            group: 'fakeValue',
+            notebookAppUrl: 'fakeValue',
+            onLayoutChange: 'fakeValue',
+            openSidebar: true, // coerced
+            query: 'fakeValue',
+            requestConfigFromFrame: 'fakeValue',
+            services: 'fakeValue',
+            showHighlights: 'fakeValue',
+            sidebarAppUrl: 'fakeValue',
+            subFrameIdentifier: 'fakeValue',
+            theme: 'fakeValue',
+            usernameUrl: 'fakeValue',
+          },
+          config
+        );
+      });
     });
   });
 
-  ['branding', 'services'].forEach(function (settingName) {
-    it(`reads ${settingName} from the host page only when in an embedded client`, function () {
-      getConfig('all', 'WINDOW');
+  describe('default values', () => {
+    beforeEach(() => {
+      // Remove all fake values
+      $imports.$mock({
+        './settings': sinon.stub().returns({
+          hostPageSetting: sinon.stub().returns(undefined),
+          annotations: undefined,
+          clientUrl: undefined,
+          group: undefined,
+          notebookAppUrl: undefined,
+          showHighlights: undefined,
+          sidebarAppUrl: undefined,
+          query: undefined,
+        }),
+      });
+    });
+
+    this.afterEach(() => {
```

Comment:
```suggestion
    afterEach(() => {
```

I assume the use of `this` here was not intended? In this case I think `this` will actually refer to the containing `context` block rather than the `describe` block due the use of arrow functions for the `describe` but function expressions for `context`.

We generally avoid the use of `this` to refer to the Mocha context in tests unless absolutely necessary - and I'm not even sure if we have ever needed it in the client.

Reply:
Correct, not intended and typo!